src/main/java/net/snowflake/client/core/QueryContextCache.java

```diff
@@ -447,5 +442,37 @@ public int hashCode() {
     public int compareTo(QueryContextElement obj) {
       return (priority == obj.priority) ? 0 : (((priority - obj.priority) < 0) ? -1 : 1);
     }
+
+    public void setId(long id) {
```

Comment:
If we pass all values of id, readTimestamp, priority, context in constructor then don't need setters also.

Reply:
In `private static QueryContextElement deserializeQueryContextElement(JsonNode node)`, function, some field might be unavailable. So I initialize an empty object and set potential fields.